### PR TITLE
Add golint and go vet to pre-commit hook

### PR DESCRIPTION
**Pull request by NegativeMjark**

### CODE REVIEW

File: hooks/pre-commit [hunk 1]
```diff
@@ -2,8 +2,9 @@
...
 set -eu
 
-golint
+golint ./...
 go fmt
+go tool vet .
 go tool vet --shadow .
-gocyclo -over 12 .
-go test
+gocyclo -over 19 .
```
Comment (Kegsay) [inline hooks/pre-commit]: Why?

File: hooks/pre-commit [hunk 2]
```diff
@@ -2,8 +2,9 @@
 
 set -eu
 
-golint
+golint ./...
 go fmt
+go tool vet .
 go tool vet --shadow .
```
Comment (Kegsay) [inline hooks/pre-commit]: uhhh... `go tool vet` is already being called?

File: hooks/pre-commit [hunk 3]
```diff
@@ -2,8 +2,9 @@
...
-gocyclo -over 12 .
-go test
+gocyclo -over 19 .
+go test .
```
Comment (Kegsay) [inline hooks/pre-commit]: The `.` shouldn't be necessary (works fine without on `gomatrix` which uses the same directory structure).

Also, you probably want to make this:
```
go test -timeout 5s -test.v
```

which sets a 5s timeout for each test and logs test names and stdout which can be useful when debugging problems

File: hooks/pre-commit (hunk 2 above)
Comment (NegativeMjark) [inline hooks/pre-commit]: --shadow seems to turn of the other checks?

Comment (Kegsay) [inline hooks/pre-commit]:
```
  -shadow
    	check for shadowed variables (experimental; must be set explicitly)
```

Comment (Kegsay) [inline hooks/pre-commit]: Oh fun.
```
By default, -all is set and all non-experimental checks are run.
For more information run
	go doc cmd/vet

Flags:
  -all
    	enable all non-experimental checks
```

So setting `-shadow` turns OFF `-all` :/

**Comment by Kegsay** [comment]
LGTM

**Pull request merged**